src/main/java/io/castle/client/api/CastleApiImpl.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    @Override
+    public CastleApi mergeContext(Object additionalContext) {
+        JsonObject contextToMerge = configuration.getModel().getGson().toJsonTree(additionalContext).getAsJsonObject();
+        JsonObject mergedContext = new ContextMerge().merge(this.contextJson, contextToMerge);
+        return new CastleApiImpl(request, doNotTrack, configuration, mergedContext);
+    }
+
+    @Override
+    public CastleApi doNotTrack(boolean doNotTrack) {
+        return new CastleApiImpl(request, doNotTrack, configuration);
+    }
+
+    @Override
+    public AuthenticateAction authenticate(String event, String userId) {
+        return authenticate(event, userId, null);
+    }
+
+    @Override
+    public AuthenticateAction authenticate(String event, String userId, Object properties) {
+        RestApi restApi = configuration.getRestApiFactory().buildBackend();
+        JsonElement propertiesJson = null;
+        if (properties != null) {
+            propertiesJson = configuration.getModel().getGson().toJsonTree(properties);
+        }
+        return restApi.sendAuthenticateSync(event, userId, contextJson, propertiesJson);
+    }
+
+    @Override
+    public void authenticateAsync(String event, String userId, @Nullable Object properties, AsyncCallbackHandler<AuthenticateAction> asyncCallbackHandler) {
+        RestApi restApi = configuration.getRestApiFactory().buildBackend();
+        JsonElement propertiesJson = null;
+        if (properties != null) {
+            propertiesJson = configuration.getModel().getGson().toJsonTree(properties);
+        }
+        restApi.sendAuthenticateAsync(event, userId, contextJson, propertiesJson, asyncCallbackHandler);
+
+    }
+
+    @Override
+    public void authenticateAsync(String event, String userId, AsyncCallbackHandler<AuthenticateAction> asyncCallbackHandler) {
+        authenticateAsync(event, userId, null, asyncCallbackHandler);
+    }
+
+    @Override
+    public void track(String event) {
+        track(event, null, null, null);
+    }
+
+    @Override
+    public void track(String event, String userId) {
+        track(event, userId, null, null);
+    }
+
+    @Override
+    public void track(String event, String userId, Object properties) {
+        track(event, userId, properties, null);
+    }
+
+    @Override
+    public void track(String event, @Nullable String userId, @Nullable Object properties, AsyncCallbackHandler<Boolean> asyncCallbackHandler) {
+        if (doNotTrack) {
+            return;
+        }
+        RestApi restApi = configuration.getRestApiFactory().buildBackend();
+        JsonElement propertiesJson = null;
+        if (properties != null) {
+            propertiesJson = configuration.getModel().getGson().toJsonTree(properties);
+        }
+        restApi.sendTrackRequest(event, userId, contextJson, propertiesJson, asyncCallbackHandler);
+    }
+
+    @Override
+    public void identify(String userId, boolean active, Object traits, Object properties) {
```

Comment:
identify shouldn't have properties

Reply:
active should be an optional param and default to true

Reply:
ok.